src/dash-table/components/ControlledTable/TooltipCursorDecorator.tsx

```diff
@@ -0,0 +1,184 @@
+            tooltips[columnId] &&
+            (
+                tooltips[columnId].length > realIndex ?
+                    tooltips[columnId][realIndex] :
+                    null
+            );
+
+        const staticTooltip = column_static_tooltip[columnId];
+
+        const conditionalTooltips = R.filter(tt => {
+            return !tt.if ||
+                ifColumnId(tt.if, columnId) ||
+                ifRowIndex(tt.if, realIndex) ||
+                ifFilter(tt.if, this.props.virtualized.data[realIndex - this.props.virtualized.offset.rows]);
+        }, column_conditional_tooltips);
+
+        return conditionalTooltips.length ?
+            conditionalTooltips[0] :
+            legacyTooltip || staticTooltip;
+    }
+
+    componentDidUpdate() {
+        const { tooltip } = this.refs as { [key: string]: HTMLElement };
+        if (!tooltip) {
+            return;
+        }
+
+        // x-axis view start / end
+        const xvs = document.body.offsetLeft;
+        const xve = document.body.offsetLeft + document.body.clientWidth;
+
+        // x-asix tooltip start / end
+        const xts = tooltip.offsetLeft;
+        const xte = tooltip.offsetLeft + tooltip.clientWidth;
+
+        // x-axis view-tooltip delta start / end
+        // positive: tooltip is in the view
+        // negative: tooltip is out of view
+        // inverted to keep the same meaning on both sides
+        const xds = xts - xvs;
+        const xde = xve - xte;
+
+        if (xde >= 0) {
+            return;
+        }
+
+        // the tooltip can only exceed the view in one direction (right)
+        // bound the adjustment to whatever is smallest (the right excess, or the left space)
+        tooltip.style.marginLeft = `${Math.max(xde, -xds)}px`;
+    }
+
+    render() {
+        let children = (this.props as any).children;
+
+        const { displayTooltip } = this.state;
+
+        let tooltip = (displayTooltip || undefined) && this.getTooltip();
+
+        let tooltipComponent;
+        if (tooltip && !this.state.md) {
+            import(/* webpackChunkName: "markdown" */ 'remarkable')
+                .then(lib => lib.default)
+                .then(Remarkable => {
+                    this.setState({ md: new Remarkable() });
+                });
+            tooltipComponent = null;
+        }
+
+        tooltipComponent = (!tooltip || !this.state.md) ?
+            null :
+            (<div
+                className='dash-table-tooltip'
+                ref='tooltip'
+                style={{
+                    left: this.state.pageX,
+                    top: this.state.pageY
+                }}
+                dangerouslySetInnerHTML={{
+                    __html: typeof tooltip === 'string' ?
+                        tooltip :
+                        tooltip.type === TooltipSyntax.Markdown ?
+                            this.state.md.render(tooltip.value) :
+                            tooltip.value
```

Comment:
For markdown yes, but do we need non-markdown tooltips to be raw HTML?

Reply:
This is not raw html, this is raw text. We do not plan on supporting raw html input / parsing

Reply:
But you're providing it to the `dangerouslySetInnerHTML` prop - so won't it be rendered as HTML?

Reply:
Right.. changing that.